source/jormungandr/jormungandr/external_services/free_floating.py

```diff
@@ -69,9 +69,27 @@ def response_marshaller(cls, response):
 
         try:
             json_response = response.json()
+            json_response = cls.build_pagination(json_response, args)
         except ValueError:
             logging.getLogger(__name__).error(
                 "impossible to get json for response %s with body: %s", response.status_code, response.text
             )
             return None
         return FreeFloatingsSerializer(json_response).data
+
+    @classmethod
+    def build_pagination(cls, resp, args):
+        # Update elements in the pagination if present in the response forseti else create
+        if resp.get('pagination') is None:
+            resp['pagination'] = {}
+        resp['pagination']['items_on_page'] = resp.get('pagination', {}).get(
+            'items_on_page', len(resp.get('free_floatings', []))
+        )
+        resp['pagination']['items_per_page'] = resp.get('pagination', {}).get(
+            'items_per_page', args.get('count')
+        )
+        resp['pagination']['start_page'] = resp.get('pagination', {}).get('start_page', args.get('start_page'))
+        resp['pagination']['total_result'] = resp.get('pagination', {}).get(
+            'total_result', len(resp.get('free_floatings', []))
+        )
```

Comment:
May be create `resp.get('pagination', {})` once instead of calling it 4 times ?

Reply:
:+1: